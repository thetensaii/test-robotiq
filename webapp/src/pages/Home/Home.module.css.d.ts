declare const styles: {
  readonly "home": string;
  readonly "title": string;
  readonly "subTitle": string;
  readonly "lift": string;
};
export = styles;

