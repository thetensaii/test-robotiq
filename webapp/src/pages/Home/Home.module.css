.home {
	display: flex;
	flex-direction: column;
	align-items: center;
	/* color: var(--primary-color); */
}

.title {
	margin: 20px;
}

.subTitle {
	margin: 10px;
}

.lift {
	margin: 15px;
}
