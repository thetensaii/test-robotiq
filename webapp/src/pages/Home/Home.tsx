import { useLifts } from 'hooks/useLifts';
import React from 'react';
import LiftCard from '../../components/LiftCard';
import styles from './Home.module.css';

export const Home: React.FC = () => {
	const [lifts] = useLifts();

	return (
		<div className={styles.home}>
			<h1 className={styles.title}>Robotiq Test</h1>
			<h2 className={styles.subTitle}>Lifts</h2>
			{lifts.map((lift) => (
				<LiftCard
					key={lift.id}
					id={lift.id}
					name={lift.name}
					currentLocation={lift.currentLocation}
					minFloor={lift.minFloor}
					maxFloor={lift.maxFloor}
				/>
			))}
		</div>
	);
};
