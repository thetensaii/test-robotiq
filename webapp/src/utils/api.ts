import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
import config from '../config';
export const apiGetRequest = async (
	endpoint: string,
	requestConfig: AxiosRequestConfig = {}
): Promise<AxiosResponse> => {
	const response: AxiosResponse = await axios.get(config.API_URL + endpoint, requestConfig);
	return response;
};
export const apiPostRequest = async (
	endpoint: string,
	data: any,
	requestConfig: AxiosRequestConfig = {}
): Promise<AxiosResponse> => {
	const response: AxiosResponse = await axios.post(
		config.API_URL + endpoint,
		data,
		requestConfig
	);

	return response;
};

export const enum ApiPaths {
	LIFTS = '/lifts',
	TRAVELS = '/travels',
}
