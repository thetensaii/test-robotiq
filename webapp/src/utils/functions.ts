export const getAllNumbersBetween = (min: number, max: number): number[] => {
	const numbers = [];

	for (let i = min; i < max + 1; i++) {
		numbers.push(i);
	}

	return numbers;
};

export const getFormData: (form: HTMLFormElement) => { [k: string]: FormDataEntryValue } = (
	form
) => {
	const formData: FormData = new FormData(form);
	return Object.fromEntries(formData);
};
