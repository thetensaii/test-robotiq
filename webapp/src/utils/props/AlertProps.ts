import { AlertMessageBoxTypes } from 'components/AlertMessageBox';

export type AlertProps = {
	type: AlertMessageBoxTypes;
	message: string;
};
