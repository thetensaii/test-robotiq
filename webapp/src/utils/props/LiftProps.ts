export type LiftProps = {
	id: number;
	name: string;
	currentLocation: number;
	minFloor: number;
	maxFloor: number;
};
