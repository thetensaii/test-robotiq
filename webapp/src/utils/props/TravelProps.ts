export type TravelProps = {
	id: number;
	liftID: number;
	sourceFloor: number;
	targetFloor: number;
};
