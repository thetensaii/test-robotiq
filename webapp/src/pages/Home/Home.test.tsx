import React from 'react';
import { render, screen } from '@testing-library/react';
import axios from 'axios';
import config from 'config';
import { ApiPaths } from 'utils/api';
import { Home } from './Home';
import { LiftProps } from 'utils/props/LiftProps';
import { act } from 'react-dom/test-utils';

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('Home page component tests', () => {
	it('should renders Home page', () => {
		mockedAxios.get.mockResolvedValueOnce({
			data: [],
			status: 200,
			statusText: '',
			headers: {},
			config: {},
		});
		let container;

		render(<Home />);

		expect(container).toMatchSnapshot();
	});

	it('should renders title', () => {
		mockedAxios.get.mockResolvedValueOnce({
			data: [],
			status: 200,
			statusText: '',
			headers: {},
			config: {},
		});

		render(<Home />);

		const titleElement = screen.getByRole('heading', { name: /robotiq test/i });
		expect(titleElement).toBeInTheDocument();
	});

	it('should renders subtitle', () => {
		mockedAxios.get.mockResolvedValueOnce({
			data: [],
			status: 200,
			statusText: '',
			headers: {},
			config: {},
		});
		act(() => {
			render(<Home />);
		});

		const titleElement = screen.getByRole('heading', { name: /lifts/i });
		expect(titleElement).toBeInTheDocument();
	});

	it('should not renders LiftCard', async () => {
		mockedAxios.get.mockResolvedValueOnce({
			data: [],
			status: 200,
			statusText: '',
			headers: {},
			config: {},
		});

		render(<Home />);

		const liftsElement = screen.queryAllByRole('button', { name: /travel/i });
		expect(mockedAxios.get).toHaveBeenCalledTimes(1);
		expect(mockedAxios.get).toHaveBeenCalledWith(config.API_URL + ApiPaths.LIFTS, {});

		expect(liftsElement.length).toBe(0);
	});

	it('should renders multiple LiftCard', async () => {
		const lifts: LiftProps[] = [
			{ id: 1, name: 'Lift a', currentLocation: 0, minFloor: -2, maxFloor: 5 },
			{ id: 2, name: 'Lift b', currentLocation: 0, minFloor: -2, maxFloor: 5 },
			{ id: 3, name: 'Lift c', currentLocation: 0, minFloor: -2, maxFloor: 5 },
		];

		mockedAxios.get.mockResolvedValueOnce({
			data: lifts,
			status: 200,
			statusText: '',
			headers: {},
			config: {},
		});

		render(<Home />);

		const liftsElement = await screen.findAllByRole('button', { name: /travel/i });

		expect(mockedAxios.get).toHaveBeenCalledTimes(1);
		expect(mockedAxios.get).toHaveBeenCalledWith(config.API_URL + ApiPaths.LIFTS, {});

		expect(liftsElement.length).toBe(3);
	});
});
