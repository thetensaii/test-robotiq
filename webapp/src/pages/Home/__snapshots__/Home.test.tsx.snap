// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component tests should renders Home page 1`] = `undefined`;
